app/presenters/user_analytics_presenter.rb

```diff
@@ -122,79 +154,181 @@ def trophy_stats
+      current_facility
+        .registered_diabetes_patients
         .group_by_period(:month, :recorded_at, last: MONTHS_AGO)
         .group(:gender)
         .count
-  end
 
-  def monthly_registrations
-    monthly_registrations_by_gender
-      .each_with_object({}) do |((date, _), count), by_date|
-        by_date[date] ||= 0
-        by_date[date] += count
-      end
+    {
+      grouped_by_gender_and_date: {
+        diabetes: {
+          follow_ups: follow_ups,
+          registrations: registrations
+        }
+      },
+
+      grouped_by_date: {
+        diabetes: {
+          follow_ups: sum_by_date(follow_ups),
+          registrations: sum_by_date(registrations)
+        }
+      },
+    }
   end
 
-  def all_time_follow_ups_by_gender
-    @all_time_follow_ups ||=
-      @current_facility
+  def all_time_total_stats
+    follow_ups =
+      current_facility
         .patient_follow_ups(:month)
+        .count
+        .values
+        .sum
+
+    registrations =
+      current_facility
+        .registered_patients
+        .count
+
+    {
+      grouped_by_date: {
+        total: {
+          follow_ups: follow_ups,
+          registrations: registrations
+        }
+      }
+    }
+  end
+
+  def all_time_htn_stats
+    follow_ups =
+      sum_by_gender(current_facility
+                      .hypertension_follow_ups(:month)
+                      .group(:gender)
+                      .count)
+
+    registrations =
+      current_facility
+        .registered_hypertension_patients
         .group(:gender)
         .count
-        .each_with_object({}) do |((_, gender), count), by_gender|
-          by_gender[gender] ||= 0
-          by_gender[gender] += count
-      end
+
+    {
+      grouped_by_gender: {
+        hypertension: {
+          follow_ups: follow_ups,
+          registrations: registrations
+        }
+      }
+    }
   end
 
-  def all_time_registrations_by_gender
-    @current_facility
-      .registered_patients
-      .group(:gender)
-      .count
+  def all_time_dm_stats
+    follow_ups =
+      sum_by_gender(current_facility
+                      .diabetes_follow_ups(:month)
+                      .group(:gender)
+                      .count)
+
+    registrations =
+      current_facility
+        .registered_diabetes_patients
+        .group(:gender)
+        .count
+
+    {
+      grouped_by_gender: {
+        diabetes: {
+          follow_ups: follow_ups,
+          registrations: registrations
+        }
+      }
+    }
   end
 
   def statistics_cache_key
-    "user_analytics/#{@current_facility.id}"
+    "user_analytics/#{current_facility.id}"
+  end
+
+  def sum_by_date(data)
+    data.each_with_object({}) do |((date, _), count), by_date|
+      by_date[date] ||= 0
+      by_date[date] += count
+    end
+  end
+
+  def sum_by_gender(data)
+    data.each_with_object({}) do |((_, gender), count), by_gender|
+      by_gender[gender] ||= 0
+      by_gender[gender] += count
+    end
```

Comment:
I'm having a hard time grokking the changes in this file. I see certain gender groupings moved around, I see some data structures (like `all_time_stats`) changing significantly. Could you help me understand what's going on here, and how it supports DM?

Reply:
Shall I walk you through it over a call?